test/collections/VectorTest.ooc

```diff
@@ -214,12 +220,27 @@ VectorTest: class extends Fixture {
 		*/
 
 		this add("nested heap vector", func {
-			sizeX := 2
-			sizeY := 3
-			heapVector := HeapVector<HeapVector<MyCover>> new(sizeX)
-			for (i in 0 .. sizeX)
-				heapVector[i] = HeapVector<MyCover> new(sizeY)
-			heapVector free()
+			sizeX := 10
+			sizeY := 10
+			additionTable := HeapVector<HeapVector<MyClass>> new(sizeX)
+			for (i in 0 .. sizeX) {
+				additionTable[i] = HeapVector<MyClass> new(sizeY)
+				for (j in 0 .. sizeY) {
+					additionTable[i][j] = MyClass new(i + j)
+				}
+			}
+			// Array access requires casting to the correct type when a collection holds a generic collection
```

Comment:
I wouldn't expect it to get fixed by Amos anytime soon, so we'd probably have to fix it ourselves.